### ISSUE TITLE
Gunicorn worker timeout causes resource name to be erroneously coalesced to 'GET'

### ISSUE THREAD

**Issue by chedrik** [DataDog/dd-trace-py#6408]
### Summary of problem
I am running a django app w/ gunicorn workers. When the worker times out for some reason, the resource name is not populated properly and is defaulting to the `GET` default method. Because of this, we are unable to identify which endpoints are timing out.

This is very similar to what is reported in https://github.com/DataDog/dd-trace-py/issues/6363, but I am more interested in the request coalescing into a generic `GET` rather than the system.exit traceback. The timeout and traceback are the same, but as long as I can get the resource name properly identified we can live with that for the time being (upgrading django / gunicorn / etc. is far out of scope to try to resolve the problem for us) 

### Which version of dd-trace-py are you using? 
1.16.0

### Which version of pip are you using?
23.1.2

### Which libraries and their versions are you using?
datadog==0.35.0
datadog-api-client==2.6.0
ddsketch==2.0.4
ddtrace==1.16.0
Django==2.2
django-filter==2.2.0
django-nose==1.4.6
django-redis==4.11.0
django-rest-swagger==2.2.0
django-storages==1.8
djangorestframework==3.10.3
gunicorn==19.9.0

### How can we reproduce your problem?
- have a gunicorn worker timeout on a request, or set request -> None in https://github.com/DataDog/dd-trace-py/blob/c6023f6c1e3f35d775622010920a7ba210994f82/ddtrace/contrib/django/patch.py#L451 and allow the code to run through the  ` utils._after_request_tags(pin, span, request, response)` code.

For whatever reason when the gunicorn worker dies, the thread doesnt exit gracefully and therefore the line `response = func(*args, **kwargs)` throws an exception, rather than being handled by django itself. Then, because the `response=None`,  the function `_set_resolver_tags(pin, span, request)` in https://github.com/DataDog/dd-trace-py/blob/c6023f6c1e3f35d775622010920a7ba210994f82/ddtrace/contrib/django/utils.py#L332 is never hit, hence the span resource name never properly set.

I am not sure why `_set_resolver_tags(pin, span, request)` is gated by `if response` since response is not an arg to the function, but an easy naive solution is to move this logic outside of that if statement.

### What is the result that you get?
- `resource_name='GET'`

### What is the result that you expected?
- `resource_name='GET /api/my-endpoint'`

### Workaround
I am able to workaround this by calling the `_set_resolver_tags(pin, span, request)` in my own middleware earlier on and manually setting the span.resource. While this seems to work, it appears that this issue should be looked in to at the ddtrace level. If maintainers are open to this issue, I am happy to work on a PR accordingly.

**Comment by tabgok** [comment]
Hi!

A PR was merged to fix this issue:  https://github.com/DataDog/dd-trace-py/pull/6674

Please take a look and let us know if this does (or doesn't) resolve this issue.

I'll close out the issue in a few days if everything looks good or there is no response.  If there's an issue after, please feel free to-reopen or create  a new issue!

**Comment by chedrik** [comment]
hey @tabgok , it will be probably 2-4 weeks before I'm able to verify this on my end due to other workload right now. Feel free to close and I will comment when I am able to verify, or re-open if needed. Thanks!
